### PR TITLE
KOGITO-9109: Jobs service complementary integration tests for KOGITO-9007

### PR DESCRIPTION
**Pull request by wmedvede**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [X] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [X] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [X] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [X] Pull Request contains link to the JIRA issue
- [X] Pull Request contains link to any dependent or related Pull Request
- [X] Pull Request contains description of the issue
- [X] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [kogito-apps|kogito-examples] tests</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [kogito-apps|kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [kogito-apps|kogito-examples] quarkus-lts</b>
 
- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [kogito-apps|kogito-examples] native</b>

- for <b>native lts checks</b>  
  Run native checks against quarkus lts branch
  Please add comment: <b>Jenkins run native-lts</b>

- for a <b>specific native lts check</b>  
  Run native checks against quarkus lts branch
  Please add comment: <b>Jenkins (re)run [kogito-apps|kogito-examples] native-lts</b>
 
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by wmedvede** [comment]
depends on: https://github.com/kiegroup/kogito-runtimes/pull/2975

**Comment by kie-ci3** [comment]
**(tests) - kogito-quarkus-examples job** `#359` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT=-Pkogito-apps-downstream
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-examples -u https://github.com/kiegroup/kogito-apps/pull/1714 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.downstream.kogito-quarkus-examples/359/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[2023-05-05T14:35:55.867Z] [INFO] Getting build-chain version from composite action file<br/>[Pipeline] readFile<br/>[Pipeline] echo<br/>[2023-05-05T14:35:55.933Z] [INFO] build-chain version recovered '^v3'<br/>[Pipeline] sh<br/>[2023-05-05T14:35:56.218Z] + npm install -g '@kie/build-chain-action@^v3' -registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/<br/>[2023-05-05T14:37:17.567Z] npm ERR! code E503<br/>[2023-05-05T14:37:17.567Z] npm ERR! 503 Service Unavailable - GET https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/@kie%2fbuild-chain-action<br/>[2023-05-05T14:37:17.567Z] <br/>[2023-05-05T14:37:17.567Z] npm ERR! A complete log of this run can be found in:<br/>[2023-05-05T14:37:17.567Z] npm ERR!     /home/jenkins/.npm/_logs/2023-05-05T14_37_05_759Z-debug.log<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>Stage "Build projects" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Sonar analysis)<br/>Stage "Sonar analysis" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Upload artifacts to repository)<br/>Stage "Upload artifacts to repository" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space after build)<br/>Stage "check space after build" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2023-05-05T14:37:18.445Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.downstream.kogito-quarkus-examples/359/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by kie-ci3** [comment]
**(tests) - serverless-workflow-examples job** `#362` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT=-Pkogito-apps-downstream
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-examples -u https://github.com/kiegroup/kogito-apps/pull/1714 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.downstream.serverless-workflow-examples/362/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[2023-05-05T14:35:56.345Z] [INFO] Getting build-chain version from composite action file<br/>[Pipeline] readFile<br/>[Pipeline] echo<br/>[2023-05-05T14:35:56.415Z] [INFO] build-chain version recovered '^v3'<br/>[Pipeline] sh<br/>[2023-05-05T14:35:56.699Z] + npm install -g '@kie/build-chain-action@^v3' -registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/<br/>[2023-05-05T14:37:18.065Z] npm ERR! code E503<br/>[2023-05-05T14:37:18.065Z] npm ERR! 503 Service Unavailable - GET https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/@kie%2fbuild-chain-action<br/>[2023-05-05T14:37:18.065Z] <br/>[2023-05-05T14:37:18.065Z] npm ERR! A complete log of this run can be found in:<br/>[2023-05-05T14:37:18.065Z] npm ERR!     /home/jenkins/.npm/_logs/2023-05-05T14_37_06_112Z-debug.log<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>Stage "Build projects" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Sonar analysis)<br/>Stage "Sonar analysis" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Upload artifacts to repository)<br/>Stage "Upload artifacts to repository" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space after build)<br/>Stage "check space after build" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2023-05-05T14:37:19.016Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.downstream.serverless-workflow-examples/362/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by kie-ci3** [comment]
**(tests) - kogito-springboot-examples job** `#373` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT=-Pkogito-apps-downstream
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-examples -u https://github.com/kiegroup/kogito-apps/pull/1714 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.downstream.kogito-springboot-examples/373/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[2023-05-05T14:35:56.571Z] [INFO] Getting build-chain version from composite action file<br/>[Pipeline] readFile<br/>[Pipeline] echo<br/>[2023-05-05T14:35:56.642Z] [INFO] build-chain version recovered '^v3'<br/>[Pipeline] sh<br/>[2023-05-05T14:35:56.929Z] + npm install -g '@kie/build-chain-action@^v3' -registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/<br/>[2023-05-05T14:37:18.285Z] npm ERR! code E503<br/>[2023-05-05T14:37:18.285Z] npm ERR! 503 Service Unavailable - GET https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/@kie%2fbuild-chain-action<br/>[2023-05-05T14:37:18.285Z] <br/>[2023-05-05T14:37:18.285Z] npm ERR! A complete log of this run can be found in:<br/>[2023-05-05T14:37:18.285Z] npm ERR!     /home/jenkins/.npm/_logs/2023-05-05T14_37_06_422Z-debug.log<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>Stage "Build projects" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Sonar analysis)<br/>Stage "Sonar analysis" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Upload artifacts to repository)<br/>Stage "Upload artifacts to repository" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space after build)<br/>Stage "check space after build" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2023-05-05T14:37:19.257Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.downstream.kogito-springboot-examples/373/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by kie-ci3** [comment]
**(tests) - kogito-apps job** `#903` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT=-Dvalidate-formatting
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-apps -u https://github.com/kiegroup/kogito-apps/pull/1714 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/903/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[2023-05-05T14:36:05.620Z] [INFO] Getting build-chain version from composite action file<br/>[Pipeline] readFile<br/>[Pipeline] echo<br/>[2023-05-05T14:36:05.688Z] [INFO] build-chain version recovered '^v3'<br/>[Pipeline] sh<br/>[2023-05-05T14:36:05.972Z] + npm install -g '@kie/build-chain-action@^v3' -registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/<br/>[2023-05-05T14:37:27.330Z] npm ERR! code E503<br/>[2023-05-05T14:37:27.330Z] npm ERR! 503 Service Unavailable - GET https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/@kie%2fbuild-chain-action<br/>[2023-05-05T14:37:27.330Z] <br/>[2023-05-05T14:37:27.330Z] npm ERR! A complete log of this run can be found in:<br/>[2023-05-05T14:37:27.330Z] npm ERR!     /home/jenkins/.npm/_logs/2023-05-05T14_37_16_371Z-debug.log<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>Stage "Build projects" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Sonar analysis)<br/>Stage "Sonar analysis" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Upload artifacts to repository)<br/>Stage "Upload artifacts to repository" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space after build)<br/>Stage "check space after build" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2023-05-05T14:37:28.153Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/903/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by wmedvede** [comment]
jenkins retest this

**Comment by kie-ci3** [comment]
**(tests) - kogito-apps job** `#905` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT=-Dvalidate-formatting -Prun-code-coverage
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-apps -u https://github.com/kiegroup/kogito-apps/pull/1714 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/905/display/redirect

**Test results:**
- PASSED: 2175
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/905/testReport/org.kie.kogito.jobs.service.management/JobServiceInstanceManagerTest/heartbeatLeader/">org.kie.kogito.jobs.service.management.JobServiceInstanceManagerTest.heartbeatLeader</a></summary>
org.mockito.exceptions.verification.TooManyActualInvocations: <br/><br/>repository.heartbeat(<br/>    JobServiceManagementInfo[id=null, lastHeartbeat=2023-05-06T08:09:36.341Z, token=5226bd46-310f-4ed5-a4b5-19116e319e04]<br/>);<br/>Wanted 1 time:<br/>-> at org.kie.kogito.jobs.service.management.JobServiceInstanceManagerTest.heartbeatLeader(JobServiceInstanceManagerTest.java:142)<br/>But was 2 times:<br/>-> at org.kie.kogito.jobs.service.management.JobServiceInstanceManager.heartbeat(JobServiceInstanceManager.java:187)<br/>-> at org.kie.kogito.jobs.service.management.JobServiceInstanceManager.heartbeat(JobServiceInstanceManager.java:187)<br/><br/><br/>	at org.kie.kogito.jobs.service.management.JobServiceInstanceManagerTest.heartbeatLeader(JobServiceInstanceManagerTest.java:142)<br/>	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/>	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)<br/>	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/>	at java.base/java.lang.reflect.Method.invoke(Method.java:566)<br/>	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)<br/>	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)<br/>	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)<br/>	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)<br/>	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)<br/>	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)<br/>	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)<br/>	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)<br/>	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)<br/>	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)<br/>	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)<br/>	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)<br/>	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)<br/>	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)<br/>	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:217)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:213)<br/>	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:138)<br/>	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)<br/>	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)<br/>	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)<br/>	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)<br/>	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)<br/>	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)<br/>	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)<br/>	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)<br/>	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)<br/>	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)<br/>	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)<br/>	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)<br/>	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)<br/>	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)<br/>	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)<br/>	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)<br/>	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)<br/>	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)<br/>	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)<br/>	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)<br/>	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)<br/>	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:188)<br/>	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)<br/>	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:128)<br/>	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)<br/>	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)<br/>	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)<br/>	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)<br/>
</details>

**Comment by wmedvede** [comment]
jenkins retest kogito-apps

**Comment by wmedvede** [comment]
Jenkins rerun kogito-apps

**Comment by wmedvede** [comment]
jenkins rerun kogito-apps tests

File: apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-common/pom.xml [hunk 1]
```diff
@@ -33,6 +33,11 @@
       <artifactId>rest-assured</artifactId>
       <scope>compile</scope>
     </dependency>
+    <dependency>
```
Comment (tiagodolphine) [inline apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-common/pom.xml:36]: why you need to add this in the data-index IT?

File: apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/pom.xml [hunk 2]
```diff
@@ -47,6 +47,16 @@
       <artifactId>rest-assured</artifactId>
       <scope>compile</scope>
     </dependency>
+    <dependency>
+      <groupId>io.rest-assured</groupId>
+      <artifactId>json-path</artifactId>
+      <scope>compile</scope>
+    </dependency>
+    <dependency>
```
Comment (tiagodolphine) [inline apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/pom.xml:55]: why do you need to add this in trusty IT?

**Review by tiagodolphine — COMMENTED** [review]
LGTM, the only thing I would suggest is instead using the prefix `Kogito9007` to use some meaningful prefix not coupled to jira id, for instance, MultipleTimerInstancesIT, or whatever... in this way it would be clear for anyone looking at these classes, etc... wdyt?

File: apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-common/pom.xml (hunk 1 above)
Comment (wmedvede) [reply apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-common/pom.xml:36]: Becasue before the rest-assured deps where taken from transitive deps, so really we had no control over which version was taken, and now, I have added the proper version in the dependencies the same as we do with assertj, mockito, etc.

File: apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/pom.xml (hunk 2 above)
Comment (wmedvede) [reply apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/pom.xml:55]: same as above

**Comment by wmedvede** [comment]
KOGITO-9109: Jobs service complementary integration tests for KOGITO-9007

**Comment by wmedvede** [comment]
@tiagodolphine suggestions where applied

File: apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-common/pom.xml (hunk 1 above)
Comment (tiagodolphine) [reply apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-common/pom.xml:36]: +1 thanks

File: apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/pom.xml (hunk 2 above)
Comment (tiagodolphine) [reply apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/pom.xml:55]: +1

**Review by tiagodolphine — APPROVED** [review]
Thanks @wmedvede

**Pull request merged**